lib/pharos/cluster_manager.rb

```diff
@@ -40,33 +40,76 @@ def validate
       addon_manager.validate
     end
 
+    # @return [Array<Pharos::Configuration::Host>]
+    def sorted_master_hosts
+      @sorted_master_hosts ||= config.master_hosts.sort_by { |h| master_sort_score(h) }
```

Comment:
I'm pretty sure that this should be an instance method on the `Pharos::Config` class?

Reply:
The `@sorted_master_hosts` memoization is also risky, because the sort scoring changes... although it would crash on the uninitialized `checks` in that case, so I guess that's good.

Reply:
I can remove memoization, it's not really needed.